### DISCUSSION TITLE
Deleting yjs xml fragment

### DISCUSSION THREAD

**Discussion by jloganolson** [ueberdosis/tiptap#1552] [General Questions]
I have a form where you can add and remove tiptap fields. The list is managed by a yjs array and it works from a user-facing standpoint, but should I be "destroying" the deleted fields' xml fragments after their IDs are removed from the yjs array? Or are they automatically cleaned up somehow?

**Comment by jloganolson** [comment]
Update: Realized I could nest the xmlFragments in a ymap (they each have a uuid) and just delete the keys there. Seems to work?

**Comment by jackbuehner** [comment]
They are not automatically cleaned, but you can delete them from the ydoc with `ydoc.share.delete(fragmentName)`.

`ydoc.share` is a JavaScript Map containing all of the shared types on the ydoc.

If you need to keep a history of deleted fields, you should store all of your fields in a ymap (as you mentioned). This will ensure that a delta containing the changes exists, which I do not think is created when directly removing a shared type with `ydoc.share.delete`
